File: solutions/python/raindrops/1/raindrops.py
sound_map = {
    3: "Pling",
    5: "Plang",
    7: "Plong",
}

def convert(number):
    result = ""
    for divisor, sound in sound_map.items():
        if number % divisor == 0:
            result += sound
    if len(result) == 0:
        result += str(number)
    return result

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import (
    "strconv"
    "strings"
)

type Sound = string

const (
    DivByThree	Sound	= "Pling"
    DivByFive	Sound	= "Plang"
    DivBySeven	Sound	= "Plong"
)

func Convert(number int) string {
	var builder strings.Builder
    if number % 3 == 0 {
        builder.WriteString(DivByThree)
    }
    if number % 5 == 0 {
        builder.WriteString(DivByFive)
    }
    if number % 7 == 0 {
        builder.WriteString(DivBySeven)
    }
    if builder.Len() == 0 {
        builder.WriteString(strconv.Itoa(number))
    }
    return builder.String()
}
